packages/async-operations/tests/test-async_operations.js

```diff
@@ -0,0 +1,383 @@
+  t.is(environmentOverrides.asyncOperationsTable, dynamoTableName);
+  t.is(environmentOverrides.lambdaName, lambdaName);
+  t.is(environmentOverrides.payloadUrl, `s3://${systemBucket}/${stackName}/async-operation-payloads/${id}.json`);
+});
+
+test.serial('The startAsyncOperation method throws error if it is unable to create an ECS task', async (t) => {
+  const createSpy = sinon.spy((obj) => ({ id: obj.id }));
+  const stubbedAsyncOperationsModel = class {
+    create = createSpy;
+  };
+
+  stubbedEcsRunTaskResult = {
+    tasks: [],
+    failures: [{ arn: randomString(), reason: 'out of cheese' }],
+  };
+  const stackName = randomString();
+
+  await t.throwsAsync(startAsyncOperation({
+    asyncOperationTaskDefinition: randomString(),
+    cluster: randomString(),
+    lambdaName: randomString(),
+    description: randomString(),
+    operationType: 'ES Index',
+    payload: {},
+    stackName,
+    dynamoTableName: dynamoTableName,
+    knexConfig: knexConfig,
+    systemBucket,
+  }, stubbedAsyncOperationsModel), {
+    instanceOf: EcsStartTaskError,
+    message: 'Failed to start AsyncOperation: out of cheese',
+  });
+});
+
+/* test.serial('The AsyncOperation start() method writes a new record to DynamoDB', async (t) => {
+  stubbedEcsRunTaskResult = {
+    tasks: [{ taskArn: randomString() }],
+    failures: [],
+  };
+  const stackName = randomString();
+
+  const { id } = await start({
+    asyncOperationTaskDefinition: randomString(),
+    cluster: randomString(),
+    lambdaName: randomString(),
+    description: randomString(),
+    operationType: 'ES Index',
+    payload: {},
+    stackName,
+    dynamoTableName: asyncOperationModel.tableName,
+    knexConfig: knexConfig,
+    systemBucket,
+
+  });
+
+  const fetchedAsyncOperation = await asyncOperationModel.get({ id });
+  t.is(fetchedAsyncOperation.taskArn, stubbedEcsRunTaskResult.tasks[0].taskArn);
+});
+ */
+test(' startAsyncOperation calls Dynamo model create method', async (t) => {
+  const createSpy = sinon.spy(() => true);
+  const stubbedAsyncOperationsModel = class {
+    create = createSpy;
+  };
+  const stackName = randomString();
+  const description = randomString();
+  const taskArn = randomString();
+
+  stubbedEcsRunTaskResult = {
+    tasks: [{ taskArn }],
+    failures: [],
+  };
+  const result = await startAsyncOperation({
+    asyncOperationTaskDefinition: randomString(),
+    cluster: randomString(),
+    lambdaName: randomString(),
+    description,
+    operationType: 'ES Index',
+    payload: {},
+    stackName,
+    dynamoTableName: dynamoTableName,
+    knexConfig: knexConfig,
+    systemBucket,
+  }, stubbedAsyncOperationsModel);
+
+  const spyCall = createSpy.getCall(0).args[0];
+
+  const expected = {
+    description,
+    operationType: 'ES Index',
+    status: 'RUNNING',
+    taskArn,
+  };
+
+  t.true(result);
+  t.deepEqual(omit(spyCall, ['id']), expected);
+  t.truthy(spyCall.id); // TODO make this introspection better?
```

Comment:
I thought we were explicitly trying to avoid proxyquire when this sort of pattern will work   Left to my own, I'll probably just remove the comment, but if you feel strongly a proxyquire'd stub is preferable I'm 'ok' with it.